### PR TITLE
feat(source-hardcoded-records): add new source

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resovling https://github.com/airbytehq/airbyte-internal-issues/issues/8471
add new source with 3 streams of hardcoded records to track performance of Airbyte Python CDK

### Follow up

- add 2 connections to our **Airbyte Internal Performance** workspace with count = 20M:
  - skinny records stream <> E2E
  - fat records stream <> E2E

## How

records are hardcoded

## Review guide

1. `airbyte-integrations/connectors/source-hardcoded-records/source_hardcoded_records/streams.py`

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-hardcoded-records/unit_tests/unit_test.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-hardcoded-records/unit_tests/unit_test.py:1]: Delete the boilerplate!

File: docs/integrations/sources/hardcoded-records.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Hardcoded Records
+
+## Sync overview
+
+The Sample Data (Hardcoded Records) source outputs sample data using the hardcoded .
```
Comment (natikgadzhi) [inline docs/integrations/sources/hardcoded-records.md:5]:
```suggestion
The Sample Data (Hardcoded Records) source outputs sample data (same record over and over again, but very fast) and is intended to be used in performance testing.
```

File: docs/integrations/sources/hardcoded-records.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+### Output schema
+
+This source will generate an "e-commerce-like" dataset with users, products, and purchases. Here's
+what is produced at a Postgres destination connected to this source:
```
Comment (natikgadzhi) [inline docs/integrations/sources/hardcoded-records.md:10]: Don't forget the schema (but ultimately I think it's fine to omit tbh).

**Review by natikgadzhi — COMMENTED** [review]
I remember @cgardens told me we've used something similar when we did platform and DB sources performance testing.

Not opposed to having a Python implementation here, but maybe we have an existing one? @davinchia / @evantahler do you know?

Re: this PR: @artem1205 can you remove the boilerplate? Realistically, we are not about to change this connector often, so I'd say the boilerplate (mostly templates) in integration_tests, acceptance-test-config and such, unit_tests — they are all not needed.

**Comment by evantahler** [comment]
We have an E2E Testing source that generates fake records, but it's Java.  We also have faker (python).  Finally, @chandlerprall made a nodejs source recently that generates fake data too!

File: airbyte-integrations/connectors/source-hardcoded-records/source_hardcoded_records/streams.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+
+from abc import ABC
+from typing import Any, Iterable, Mapping
+
+from airbyte_cdk.sources.streams import Stream
+
+
+class HardcodedStream(Stream, ABC):
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-hardcoded-records/source_hardcoded_records/streams.py:10]: I would like to see a stream with 24 million rows of exactly this schemas and values:
```
{ field1: "valuevaluevaluevaluevalue1", 
field2: "valuevaluevaluevaluevalue1", 
field3: "valuevaluevaluevaluevalue1", 
field4: "valuevaluevaluevaluevalue1",
field5: "valuevaluevaluevaluevalue1"
}
```

this is the format we use for all of our other performance testing. doing the exact same thing helps make it comparable to other tests

File: docs/integrations/sources/hardcoded-records.md (hunk 3 above)
Comment (artem1205) [reply docs/integrations/sources/hardcoded-records.md:10]: updated

File: airbyte-integrations/connectors/source-hardcoded-records/source_hardcoded_records/streams.py (hunk 4 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-hardcoded-records/source_hardcoded_records/streams.py:10]: added

File: airbyte-integrations/connectors/source-hardcoded-records/unit_tests/unit_test.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-hardcoded-records/unit_tests/unit_test.py:1]: deleted

File: airbyte-integrations/connectors/source-hardcoded-records/source_hardcoded_records/streams.py (hunk 4 above)
Comment (cgardens) [reply airbyte-integrations/connectors/source-hardcoded-records/source_hardcoded_records/streams.py:10]: makes sense. thank you!

**Review by natikgadzhi — APPROVED** [review]
Approving to unblock, use your best judgement. I strongly suggest that you delete the `acceptance-test-config.yml`, `integration_tests`, and `schemas` if they are only used for tests.

Basically, this is a dev tool that is unlikely to change much.

**Pull request merged**